src/app/components/LanguageInfo.svelte

```diff
@@ -18,7 +18,7 @@
   // prevent triggering compilation twice
   let listenForMetaDataChanges: boolean = false;
 
-  onMount(async () => {
+  export async function getMetadata() {
```

Comment:
This is an unusual function to export, especially since it has **side-effects** on the component — namely, if the storedProjectData has changed, then the component will re-render, and also start listening for metadata changes.

It seems to violate the principle of [Command-query separation](https://en.wikipedia.org/wiki/Command%E2%80%93query_separation).

What is the purpose of exporting this function? Can the fundamental functionality needed by callers be handled by a different function? Can there be an Extract Function refactoring that can be used here to isolate the component from side-effects?

Reply:
The reason for exporting is because savedProjectData changes after importing  and so the languageInfo component will need to `fetchAllCurrentProjectMetadata()`. It's called once on mount and then needs to be triggered again by the parent component after a new project is imported, the export is so that it can be called again from Languages.

Reply:
I see what you're saying about the `listenForMetaDataChanges`. I will now set it as false when `getMetadata()` is called so that compilation is not triggered multiple times while setting the languageInfo to the stored/imported value.

Reply:
I should have been more explicit: It's _very poor practice_ to name a function "getX" that, in addition to getting X, also does side-effects such as triggering a re-render.

Svelte's reactive assignments ensure that any assignments to component-scope variables, such as `authorName`, `copyright`, etc. will trigger a DOM update: https://svelte.dev/tutorial/reactive-assignments
 

I'm also _very_ concerned about this line:

> It's called once on mount and then **needs to be triggered again** by the parent component after a new project is imported, the export is so that it can be called again from Languages.

A contract like "the parent needs to trigger the re-rendering again of the child is error-prone; even if it were _absolutely required_ to do it this way, such behaviour must be documented. That said, I think an easier soultion is for both elements to subscribe to stores instead, and let Svelte decide when re-rendering should take place.